packages/core/src/lib/streamManager/chainStreamManager.ts

```diff
@@ -0,0 +1,111 @@
+import { streamAIResponse } from './step/streamAIResponse'
+import { Message as LegacyMessage } from '@latitude-data/constants/legacyCompiler'
+import { Chain } from 'promptl-ai'
+import {
+  renderChain,
+  ValidatedChainStep,
+} from '../../services/chains/ChainValidator'
+import { StreamManager, StreamManagerProps } from '.'
+import { resolveToolsFromConfig } from './resolveTools'
+import { CachedApiKeys } from '../../services/chains/run'
+
+/**
+ * ChainStreamManager extends StreamManager to handle streaming for multi-step AI chains.
+ *
+ * Key differences from DefaultStreamManager:
+ * - Handles sequential steps in a Chain, recursively calling step() to advance through the chain
+ * - Manages chain state and context between steps
+ * - Renders each step of the chain dynamically before execution
+ * - Supports chain completion detection to end streaming when the chain is finished
+ * - Uses providers from a providersMap rather than a single provider
+ */
+export class ChainStreamManager extends StreamManager implements StreamManager {
```

Comment:
Does it need to use `implements` if it is extending from it?

Reply:
yes because it implements an abstract method

Reply:
But why? Aren't these expected errors?